mcu/src/parser.c

```diff
@@ -0,0 +1,64 @@
+// #include "pmt_config.h"
+#include "parser.h"
+#include "string.h"
+#include "ctype.h"
+
+// from here: https://stackoverflow.com/questions/9907160/how-to-convert-enum-names-to-string-in-c
+const char const pmt_type_str[] = 
+{
+    [SIGNAL] = 's',
+    [TRIGGER] = 't'
+};
+
+bool parse_command(const char *input, pmt_data_t *parsed)
+{
+    // validate command format
+    if(!validate_format(input))
+    {
+        return false;
+    }
+
+    // extract PMT type and voltage value
+    if(input[0] == pmt_type_str[SIGNAL])
+    {
+        parsed->pmt = SIGNAL;
+    } else {
+        parsed->pmt = TRIGGER;
+    }
+    
+    const char* number = &input[1];
+    uint16_t voltage = atoi(number);
+    parsed->data = voltage;
+
+    // if all is well, store the PMT type and voltage in the parsed structure and return true,
+    // otherwise, return false to tell the caller that the command isn't valid
+    //uint16_t bound = pmts[parsed->pmt].control_voltage_upper_bound;
+    if(!validate_voltage_bounds(voltage,pmts[parsed->pmt].control_voltage_upper_bound,pmts[parsed->pmt].control_voltage_lower_bound))
+    {
+        return false;
+    }
+
+    return true;
+}
```

Comment:
Consider changing this. This if statement duplicates the `validate_voltage_bounds()` function. Some options:
1. Fold the `validate_voltage_bounds()` function into the `parse_command()` function.
2. Change the control flow in `parse_command()` so we aren't early-returning